server/storage/backend/backend.go

```diff
@@ -164,9 +164,17 @@ func New(bcfg BackendConfig) Backend {
 	return newBackend(bcfg)
 }
 
-func NewDefaultBackend(lg *zap.Logger, path string) Backend {
-	bcfg := DefaultBackendConfig(lg)
-	bcfg.Path = path
+func NewDefaultBackend(bc BackendConfig) Backend {
+
+	bcfg := DefaultBackendConfig(bc.Logger)
+	bcfg.Path = bc.Path
+
+	if bc.MmapSize > 0 {
+		bcfg.MmapSize = bc.MmapSize
+	} else {
+		bcfg.MmapSize = InitialMmapSize
+	}
```

Comment:
There are several fields in `BackendConfig`, but we we make use of only `MmapSize`, then why we have to pass into a `bc BackendConfig`? Suggest to follow the similar pattern something like below. We can only define one function `WithMmapSize` for now, and add others when needed in future,

https://github.com/etcd-io/etcd/blob/42f0cb9762cafa440a3f77884b0deb454ccb22c5/client/v3/op.go#L450-L477

Reply:
To make sure I re-implement correctly, you're suggesting to also return the `NewDefaultBackend` signature to `*zap.Logger, string`, right?

Reply:
Something like this:
https://github.com/etcd-io/etcd/blob/9ecdead32e5911f79a14613e8cbca5ef28866352/server/storage/backend/backend.go#L167-L181

And used like this:
https://github.com/etcd-io/etcd/blob/9ecdead32e5911f79a14613e8cbca5ef28866352/etcdutl/snapshot/v3_snapshot.go#L328

Reply:
Exactly.

Reply:
Done.